test/server.js

```diff
@@ -445,6 +454,65 @@ describe("server", function() {
                 });
             }
         );
+        it("attempts to upsert 'new_message' notification for each participant if message is successfully added",
+            function() {
+                var beforeTimestamp;
```

Comment:
This test is quite big. Can you refactor it to make it easier to read how its performing the test?